File: lsd-interceptors/src/main/java/io/lsdconsulting/interceptors/messaging/HtmlRenderer.java
package io.lsdconsulting.interceptors.messaging;

import org.springframework.messaging.MessageHeaders;

import static j2html.TagCreator.*;
import static java.lang.System.lineSeparator;
import static java.util.stream.Collectors.joining;
import static org.apache.commons.lang3.StringUtils.isEmpty;

public class HtmlRenderer {
    public static String renderHtmlFor(MessageHeaders messageHeaders, String prettyBody) {
        return div(
                section(h3("Message Headers"), p(prettyPrintHeaders(messageHeaders))),
                isEmpty(prettyBody) ? p() : section(h3("Body"), p(prettyBody))
        ).render();
    }

    private static String prettyPrintHeaders(MessageHeaders requestHeaders) {
        return requestHeaders.entrySet().stream().map(entry ->
                        entry.getKey() + ": " + entry.getValue())
                .collect(joining(lineSeparator()));
    }
}

File: lsd-interceptors/src/main/kotlin/io/lsdconsulting/interceptors/messaging/HtmlRenderer.kt
package io.lsdconsulting.interceptors.messaging

import j2html.TagCreator
import org.springframework.messaging.MessageHeaders

fun renderHtmlFor(messageHeaders: MessageHeaders, prettyBody: String?): String {
    return TagCreator.div(
        TagCreator.section(TagCreator.h3("Message Headers"), TagCreator.p(prettyPrintHeaders(messageHeaders))),
        if (prettyBody.isNullOrEmpty()) TagCreator.p()
        else TagCreator.section(
            TagCreator.h3("Body"),
            TagCreator.p(prettyBody)
        )
    ).render()
}

private fun prettyPrintHeaders(requestHeaders: MessageHeaders): String =
    requestHeaders.entries.joinToString(separator = System.lineSeparator()) { (key, value): Map.Entry<String, Any> ->
        "$key: $value"
    }
